presto-docs/src/main/sphinx/functions/geospatial.rst

```diff
@@ -64,6 +87,14 @@ Constructors
 
     Returns a geometry type polygon object from WKT representation.
 
+.. function:: to_spherical_geography(Geometry) -> SphericalGeography
+
+    Converts a geometry value defined in (longitude, latitude) space to a geometry on the sphere of the Earth's radius.
```

Comment:
The semantics of this function are not obvious. It's clear how it'd behave for points: it treats each pair as a lat/lon coordinate so it just constructs a SphericalGeography based on those points on the surface of the sphere. It's not clear what that does to lines or other shapes. 

In fact, it's strange to say that the input can be a line with ends defined using lat/lon, since those imply a spherical coordinate system, but the input is a geometry, not a SphericalGeography.

Reply:
@martint Martin, are you saying that the docs need to clarify that lines are assumed to be great-circle arcs between start and end points?

Reply:
I think the semantic are fundamentally awkward due to the fact that entities described by geometries (line, polygon, curve, etc) are on a plane. It's easy to see how a point, a polygon or a line could be reinterpreted as being on a the surface of sphere, but what about more complex geometries like curves, circles, NURBS curves, spirals, or any of the 3D geometries defined by the spec? 

Clarifying the docs would help, but we also need to make sure the function is strict about which geometries it supports.

Reply:
@martint ah, got your point. @mbasmanova do the Presto Geospatial functions support 3D geometries or geometries defined by mathematical curve functions rather than discrete point sequences? The current documentation https://prestodb.io/docs/current/functions/geospatial.html seems to imply that only 2D geometries compatible with WKT are supported and I wonder if that's already constrained by the Geometry constructors. We can further clarify that in both the Geometry documentation and SphericalGeography documentation.

Reply:
@gaoxm Ming, you are correct. Currently Presto supports only a subset of 2D geometries, e.g. point, line-string, polygon, multi-point, multi-line-string, multi-polygons and geometry collections made of these geometries.

Reply:
@mbasmanova cool. If the Geometry constructor already restricts the types of geometries allowed, then it seems unnecessary to add the restriction logics to to_spherical_geography() for now. @martint what do you think? We can for sure add more clarification about the supported geometry types in the documentation for both Geometry and SphericalGeography. Does that sound alright?

Reply:
It needs to be explicitly validated by this function. Otherwise, when we add support for other geometries, this function will silently start doing the wrong thing.

Reply:
@martint 👍cool, I'll look to add necessary checks.

Reply:
@martint Martin,

> It's easy to see how a point, a polygon or a line could be reinterpreted as being on a the surface of sphere, but what about more complex geometries like curves, circles, NURBS curves, spirals, or any of the 3D geometries defined by the spec?

I think any 2D geometry can be mapped to a sphere. Since a geometry is a set of points and every 2D point can be mapped to a sphere, the whole geometry can be mapped. Hence, the mapping is not all that interesting, however, the functions that compute something (length, area, distance) might be tricky to define and implement in a sensible way. Hence, it seems to me that this function could apply to any 2D geometry, but functions that accept spherical geometry may need to restrict themselves to certain types.

3D geometries don't map to a sphere indeed. 

That said, having this function check for dimension and/or type of geometry is expensive as it requires it to deserialize the whole object. Therefore, it seems like skipping the validation is a more sensible choice right now.

What do you think?